mongodb/intro/date-aggregation/aggregation-with-date-expression-operators-iii.md

```diff
@@ -0,0 +1,172 @@
+---
+author: Stefan-Stojanovic
+
+aspects:
+  - introduction
+  - new
+
+type: normal
+
+category: how to
+
+---
+
+# Aggregation With `$hour`, `$minute`, `$second` and `$millisecond`
+
+---
+## Content
+
+For the document used in aggregation, make sure to check the Previous Document[1] footnote.
+
+### $hour
+
+The `$hour` date operator is used to extract the exact hour the document was created on. The output goes from 0 to 23.
+
+Example:
+```js
+db.pokedex.aggregate([
+  { $match: { name: "Bulbasaur" } },
+  {
+    $project: {
+      createdAtHour: { $hour: "$createdAt" },
+      _id: 0
+    }
+  }
+]);
+
+```
+
+Output:
+```js
+{ 
+  "createdAtHour" : 7 
+}
+```
+
+### $minute
+
+The `$minute` date operator is used to extract the exact minute the document was made on. The output goes from 0 to 59.
+
+Example:
+```js
+db.pokedex.aggregate([
+  { $match: { name: "Bulbasaur" } },
+  {
+    $project: {
+      createdAtMinute: { $minute: "$createdAt" },
+      _id: 0
+    }
+  }
+]);
+
+```
+Output:
+```js
+{ 
+  "createdAtMinute" : 21 
+}
+```
+
+### $second
+
+The `$second` date operator is used to extract the exact second the document was made on. Although the output goes from 0 to 59, it is possible for the output to be 60 in case of a Leap Second[2].
+
+Example:
+```js
+db.pokedex.aggregate([
+  { $match: { name: "Bulbasaur" } },
+  {
+    $project: {
+      createdAtSecond: { $second: "$createdAt" },
+      _id: 0
+    }
+  }
+]);
+
+```
+Output:
+```js
+{ 
+  "createdAtSecond" : 14 
+}
+```
+
+### $millisecond
+
+The `$millisecond` date operator is used to extract the exact millisecond the document was made on. The output goes from 0 to 999.
+
+Example:
+```js
+db.pokedex.aggregate([
+  { $match: { name: "Bulbasaur" } },
+  {
+    $project: {
+      createdAtMillisecond: {
+        $millisecond: "$createdAt"
+      },
+      _id: 0
+    }
+  }
+]);
+
+```
+
+Output:
+```js
+{ 
+  "createdAtMillisecond" : 0 
+}
+```
+
+---
+## Practice
```

Comment:
we can split this question and create a revision q from the leftovers.

my suggestion is:
- for PQ we can have the questions like: "Which operator is used to extract seconds/minutes/hours"
- for RQ we can have the questions like: "The result of the $minute operator is between 0-59`